### ISSUE TITLE
selecting multiple image in a single app by multiple buttons

### ISSUE THREAD

**Issue by subhamoychatterjee** [chaquo/chaquopy#426]
Hi, Malcom.
I am a licenced user of the Chaquopy SDK. I have found a limitation that once the app is run, it must be closed completely before running the second time. Otherwise, it will not show the final result in the textview. Somehow, I have managed when working with a single image.
But presently, my app development needs selecting at least two images from the gallery with two different buttons. Comparing the results of the two images I am taking a decision. The second image selection by the button is not working. I am assuming that the SDK is assuming I am running the app without closing it. I am also using two python scripts for it.

Is it possible to work with multiple images, multiple python files  in chaquopy? What is the alternative? This work is very urgent.
Roughly, I have to select two images two times. Do two different image processing operations and comparing these, final results will be taken.
Please help me regarding this.

the code snippet is given below:

```
case SELECT_PHOTO2:
                if (resultCode == RESULT_OK) {
                    Uri selectedImage = imageReturnedIntent.getData();

                    InputStream imageStream = null;
                    try {
                        imageStream = getContentResolver().openInputStream(selectedImage);
                    } catch (FileNotFoundException e) {
                        e.printStackTrace();
                    }
                    Bitmap yourSelectedImage = BitmapFactory.decodeStream(imageStream);
                    image.setImageBitmap(yourSelectedImage);
                    image.setImageURI(selectedImage);// To display selected image in image view

                    //for chaquopy python
                    //conversion of bitmap to stringimage by imagestream or I can directly use my imagestream
                    String imageString=getStringImage(yourSelectedImage);
//then call a python script lamp.py. it is showing the output properly in the textview

//activity on new button after completion of the previous activity

case SELECT_PHOTO22:
                if (resultCode == RESULT_OK) {
                    Uri selectedImage = imageReturnedIntent.getData();

                    InputStream imageStream = null;
                    try {
                        imageStream = getContentResolver().openInputStream(selectedImage);
                    } catch (FileNotFoundException e) {
                        e.printStackTrace();
                    }
                    Bitmap yourSelectedImage = BitmapFactory.decodeStream(imageStream);
                    //commented it to check whether chaquopy can handle multiple or not
                    //image.setImageBitmap(yourSelectedImage);
                    //image.setImageURI(selectedImage);// To display selected image in image view

                    //for chaquopy python
                    //conversion of bitmap to stringimage by imagestream or I can directly use my imagestream
                    String imageString=getStringImage(yourSelectedImage);
// call another pythonscript lampref.py.
```

***particularly this portion is not showing anything in textview.

Is it possible to work with multiple images in Chaquopy? I need the second image operation after completing the first one.

**Comment by subhamoychatterjee** [comment]
If I comment either of this button activity and use a single activity, everything is fine. It is suggesting that in an app I can do image selection and image operation just once. Then closing the app, I have to start again.

But my present development demands for multiple image selection and different operations one after the other, how to get rid of that problem?

**Comment by mhsmith** [comment]
I'm not sure why you think the problem is with Chaquopy, because you haven't included any code which actually uses Chaquopy.

I don't know anything about your app, so I can't really answer high-level questions like "why isn't it showing anything in the textview". You need to narrow it down to something like "why is this method returning this value". You can do this using a [debugger](https://developer.android.com/studio/debug) or [log messages](https://developer.android.com/studio/debug/am-logcat.html).

**Comment by subhamoychatterjee** [comment]
Actually I did a silly mistake. When I was calling my second python script, I used if ( ! Python.isStarted()) {. I changed it to if ( Python.isStarted()) {. I am very new to this SDK. Every mistake is a lesson for me. 
For running an app sponteniously (without closing it), I tried the same tricks. if ( ! Python.isStarted()) { and if ( Python.isStarted()) {. Now it is fine.
Thank you very much for your support and such a wonderful SDK.
